src/hypothesis/searchstrategy/strategies.py

```diff
@@ -361,7 +361,7 @@ def validate(self):
             self.do_validate()
             self.is_empty
             self.has_reusable_values
-        except:
+        except Exception:
```

Comment:
The rest bare except's turned into `expect BaseException`, but not here. There is some reason for that?

Reply:
Yep - the others are wrapping arbitrary user code, which could raise anything.  Because the validation code is ours, we can know catching Exception is sufficient.

TBH one or two of the `except BaseException:` cases should probably catch `Exception` too, but I decided to err on the safe side and stick with existing behaviour.